javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/HelperInjector.java

```diff
@@ -30,11 +31,23 @@
 import net.bytebuddy.dynamic.DynamicType;
 import net.bytebuddy.dynamic.loading.ClassInjector;
 import net.bytebuddy.utility.JavaModule;
+import org.checkerframework.checker.nullness.qual.NonNull;
 import org.checkerframework.checker.nullness.qual.Nullable;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-/** Injects instrumentation helper classes into the user's classloader. */
+/**
+ * Injects instrumentation helper classes into the user's classloader.
+ *
+ * <p>Care must be taken when using this class. It is used both by the javaagent during its runtime
+ * and by gradle code generation plugins during build time. And some code paths in this class
```

Comment:
Wait, this class is used by the codegen plugin? I was aware that it was being used by the muzzle check plugin, but the codegen?

Reply:
Hm, you are right. I conflated those two. Will fix the docs.